File: src/main/java/uk/laxd/dndSimulator/equipment/UnarmedAttack.java
package uk.laxd.dndSimulator.equipment;

import uk.laxd.dndSimulator.ability.Ability;
import uk.laxd.dndSimulator.action.MeleeAttackAction;
import uk.laxd.dndSimulator.action.DamageType;
import uk.laxd.dndSimulator.character.Character;
import uk.laxd.dndSimulator.dice.Die;

import java.util.Collection;
import java.util.Collections;

public class UnarmedAttack extends Weapon {
    @Override
    public int getToHitModifier(MeleeAttackAction attackAction) {
        Character attacker = attackAction.getPerformer();
        return attacker.getAbilityModifier(Ability.STRENGTH) + attacker.getProficiencyBonus();
    }

    @Override
    public Collection<Die> getDamageDice(MeleeAttackAction attackAction) {
        return Collections.emptyList();
    }

    @Override
    public int getDamage(MeleeAttackAction attackAction) {
        return attackAction.getPerformer().getAbilityModifier(Ability.STRENGTH) + 1;
    }

    @Override
    public DamageType getDamageType() {
        return DamageType.BLUDGEONING;
    }
}

File: app/src/main/kotlin/uk/laxd/dndSimulator/equipment/UnarmedAttack.kt
package uk.laxd.dndSimulator.equipment

import uk.laxd.dndSimulator.action.MeleeAttackAction
import uk.laxd.dndSimulator.ability.Ability
import uk.laxd.dndSimulator.dice.Die
import uk.laxd.dndSimulator.action.DamageType

class UnarmedAttack : Weapon("Unarmed attack") {
    override var toHitModifier = 0

    override fun getDamageDice(attackAction: MeleeAttackAction): MutableList<Die> {
        return mutableListOf()
    }

    override fun getDamage(attackAction: MeleeAttackAction): Int {
        return attackAction.actor.getAbilityModifier(Ability.STRENGTH) + 1
    }

    override val damageType = DamageType.BLUDGEONING
    override val range = 5
    override val priority = 0.01
    override fun getProficiencyNames(): List<String> = listOf()
}